app/models/security_requirements_guide.rb

```diff
@@ -51,4 +54,32 @@ def self.latest
   def full_title
     "#{title} #{version}"
   end
+
+  def parsed_benchmark
+    @parsed_benchmark ||= Xccdf::Benchmark.parse(xml)
+  end
+
+  attr_writer :parsed_benchmark
+
+  private
+
+  def import_srg_rules
+    srg_rules = parsed_benchmark.rule.map do |rule|
+      SrgRule.from_mapping(rule, id)
+    end
+
+    # Examine import results for failures
+    success = SrgRule.import(srg_rules, all_or_none: true, recursive: true).failed_instances.blank?
+    if success
+      reload
+    else
+      errors.add(:base, 'Some rules failed to import successfully for the SRG.')
```

Comment:
So I ran into this when trying to add a new component based on an existing SRG (before the migrations of this PR). I was creating the component off of `Application Server Security Requirements Guide` which I've done before and I wasn't sure if this problem has to do with the fact that this SRG was created before/after the migration, so I tried to remove that existing SRG and upload it again and ran into this when trying to remove it (which I assume is because there were still components associated with it). Am I correct in thinking that we just need to handle the error by having the front end just display a message saying that it can't be removed because it still has components referencing it?
![Screen Shot 2021-11-01 at 5 18 33 PM](https://user-images.githubusercontent.com/9956561/139743968-a7e9af1e-98ad-486d-8e06-b6302c5c49b7.png)

Reply:
It looks as though the error here (at least in the screenshot) is due to `@project_member` being referenced when it should be @srg`. I think it will then display the correct error.